File: chapter1/1.1-is-unique/unique.py
from __future__ import print_function
import json

def read_json(file_path):
	with open(file_path, 'r') as file:
		data = json.load(file)
	return data

def check_unique(string):
	if len(string) <= 1:
		return True
	sorted_str = sorted(string)
	next_char = sorted_str[1]
	for char in sorted_str:
		if char == next_char:
			return False
		next_char = char
	return True


def check_unique_bit(string):
	check = 0
	for char in string:
		aux = ord(char) - ord('a')
		if check & 1 << aux > 0:
			return False
		check = check | 1 << aux
	return True

data = read_json("sample.json")
for i in data.values():
	print(check_unique(i))
	print(check_unique_bit(i))

File: chapter1/1.1-is-unique/unique.js
var json_sample = require("./sample.json");

function hasUniqueChars(str) {
	if (str.length > 256) {
		return false;
	}
	else if (str.length <= 1) {
		return true;
	}
	else for (var i = 0; i < str.length; i++) {
		if (str.indexOf(str[i]) !== str.lastIndexOf(str[i]) || str.lastIndexOf(str[i]) !== i) {
			return false;
		}
	}
	return true;
}

for (var key in json_sample){
	console.log(hasUniqueChars(json_sample[key]));
}
